browser-test/src/application_navigation.test.ts

```diff
@@ -1438,5 +1438,161 @@ describe('Applicant navigation flow', () => {
+    const questionText1 = 'text-test-q-one'
+    const questionText2 = 'text-test-q-two'
+    const screen1 = 'Screen 1'
+    const screen2 = 'Screen 2'
+    const screen3 = 'Screen 3'
+
+    beforeAll(async () => {
+      const {page, adminQuestions, adminPrograms, adminPredicates} = ctx
+      await loginAsAdmin(page)
+      await enableFeatureFlag(page, 'esri_address_correction_enabled')
+
+      // Create Questions
+      await adminQuestions.addAddressQuestion({
+        questionName: questionAddress,
+        questionText: questionAddress,
+      })
+
+      await adminQuestions.addTextQuestion({
+        questionName: questionText1,
+        questionText: questionText1,
+      })
+
+      await adminQuestions.addTextQuestion({
+        questionName: questionText2,
+        questionText: questionText2,
+      })
+
+      // Create Program
+      await adminPrograms.addProgram(programName)
+
+      // Attach questions to program
+      await adminPrograms.editProgramBlock(programName, screen1, [
+        questionAddress,
+      ])
+
+      await adminPrograms.addProgramBlock(programName, screen2, [questionText1])
+
+      await adminPrograms.addProgramBlock(programName, screen3, [questionText2])
+
+      await adminPrograms.goToBlockInProgram(programName, screen1)
+
+      await adminPrograms.clickAddressCorrectionToggleByName(questionAddress)
+
+      const addressCorrectionInput =
+        adminPrograms.getAddressCorrectionToggleByName(questionAddress)
+
+      // TODO: Change this to true when done with other PR 6598 before merging this PR
+      expect(await addressCorrectionInput.inputValue()).toBe('false')
+
+      // Set thing to soft eligibilty
+      await adminPrograms.toggleEligibilityGating()
+
+      // Add address eligibility predicate
+      await adminPrograms.goToEditBlockEligibilityPredicatePage(
+        programName,
+        screen1,
+      )
+
+      await adminPredicates.addPredicates([
+        {
+          questionName: questionAddress,
+          scalar: 'service_area',
+          operator: 'in service area',
+          values: ['Seattle'],
+        },
+      ])
+
+      // Add the address visibility predicate
+      await adminPrograms.goToBlockInProgram(programName, screen2)
+
+      await adminPrograms.goToEditBlockVisibilityPredicatePage(
+        programName,
+        screen2,
+      )
+
+      await adminPredicates.addPredicates([
+        {
+          questionName: questionAddress,
+          action: 'shown if',
+          scalar: 'service_area',
+          operator: 'in service area',
+          values: ['Seattle'],
+        },
+      ])
+
+      // Publish Program
+      await adminPrograms.gotoAdminProgramsPage()
+      await adminPrograms.publishProgram(programName)
+
+      await logout(page)
+    })
+
+    it('when address is eligible show hidden screen', async () => {
+      const {page, applicantQuestions} = ctx
+      await enableFeatureFlag(page, 'esri_address_correction_enabled')
```

Comment:
question (non-blocking): do you need to enable the feature flag again since you enable it in the `beforeAll`?

Reply:
Given it's running in the same textcontext probably not.